### PR TITLE
oauth2/integration: better library integration

### PR DESCRIPTION
**Pull request by mlesikov**
The library had a strog dependency to the HttpServlet. All the endpoint
setups were configured in the OAuth2Servlet, so when we try to integrate
the oauth2 library in our code we should use this servlet. This makes
the integration a litle bit restrictive and cause troubles if would like
to use frameworks like SparkJava.
Now the configurations logic was moved out and separated from the need to
use directly the HttpServlet in your project

SparkJava integration example : 
```kotlin
fun main(args: Array<String>) {

    val tokens = null
    val jwtKeyStore = null
    val keyStore = null
    val identityFinder = null
    val resourceOwnerIdentityFinder = null
    val clientAuthorizationRepository = null
    val clientFinder = null
    val oauth2Config  = OAuth2Config.newConfig()
                .tokens(tokens)
                .jwtKeyStore(jwtKeyStore)
                .keyStore(keyStore)
                .identityFinder(identityFinder)
                .resourceOwnerIdentityFinder(resourceOwnerIdentityFinder)
                .clientAuthorizationRepository(clientAuthorizationRepository)
                .clientFinder(clientFinder)
                .loginPageUrl("/ServiceLogin?continue=")
                .build()
    val oauth2ServletApi = Oauth2ServlertApiSupportFactory().create(oauth2Config)

    Spark.path("/o/oauth2/v1/") {
        Spark.get("*", Oauth2Handler(oauth2ServletApi))
        Spark.post("*", Oauth2Handler(oauth2ServletApi))
    }

    println("App is up and running")
}

class Oauth2Handler(private val oauth2ServletApi: Oauth2ServlertApiSupport) : Route {

    override fun handle(request: Request?, response: Response?): Any? {

        oauth2ServletApi.serve(request!!.raw(), response!!.raw())

        return response.raw()
    }
}
```

### CODE REVIEW

File: oauth2-server/src/main/java/com/clouway/oauth2/Oauth2ServlertApiSupportFactory.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+import com.google.common.base.Optional;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+/**
+ * @author Mihail Lesikov (mihail.lesikov@clouway.com)
+ */
+public class Oauth2ServlertApiSupportFactory {
```
Comment (mgenov) [inline oauth2-server/src/main/java/com/clouway/oauth2/Oauth2ServlertApiSupportFactory.java]: typo: OAuth2Servlet

File: oauth2-server/src/main/java/com/clouway/oauth2/Oauth2ServlertApiSupport.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package com.clouway.oauth2;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+/**
+ * @author Mihail Lesikov (mihail.lesikov@clouway.com)
+ */
+public interface Oauth2ServlertApiSupport {
```
Comment (mgenov) [inline oauth2-server/src/main/java/com/clouway/oauth2/Oauth2ServlertApiSupport.java]: typo: OAuth2ServletApiSupport

**Review by mgenov — CHANGES_REQUESTED** [review]

File: oauth2-server/src/main/java/com/clouway/oauth2/Oauth2ServlertApiSupportFactory.java (hunk 1 above)
Comment (mlesikov) [reply oauth2-server/src/main/java/com/clouway/oauth2/Oauth2ServlertApiSupportFactory.java]: fixed

File: oauth2-server/src/main/java/com/clouway/oauth2/Oauth2ServlertApiSupport.java (hunk 2 above)
Comment (mlesikov) [reply oauth2-server/src/main/java/com/clouway/oauth2/Oauth2ServlertApiSupport.java]: fixed

File: oauth2-server/pom.xml [hunk 3]
```diff
@@ -4,11 +4,11 @@
   <parent>
     <groupId>com.clouway.security</groupId>
     <artifactId>parent</artifactId>
-    <version>1.0.6-SNAPSHOT</version>
+    <version>1.0.7-SNAPSHOT</version>
```
Comment (mgenov) [inline oauth2-server/pom.xml]: Dumping happens automatically and this change is not required.

Comment (mlesikov) [inline oauth2-server/pom.xml]: I've changed and the version of the library

Comment (mgenov) [reply oauth2-server/pom.xml]: http://maven.apache.org/maven-release/maven-release-plugin/

Comment (mlesikov) [reply oauth2-server/pom.xml]: ok I'll roll it back

File: oauth2-server/src/main/java/com/clouway/oauth2/OAuth2ServletApiSupport.java [hunk 4]
```diff
@@ -0,0 +1,14 @@
+package com.clouway.oauth2;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+/**
+ * @author Mihail Lesikov (mihail.lesikov@clouway.com)
+ */
+public interface OAuth2ServletApiSupport {
```
Comment (VasilMitovClouway) [inline oauth2-server/src/main/java/com/clouway/oauth2/OAuth2ServletApiSupport.java]: Maby just OAuth2ApiSupport so that Servlet is not involved.

Comment (mlesikov) [reply oauth2-server/src/main/java/com/clouway/oauth2/OAuth2ServletApiSupport.java]: sounds good to me

Comment (mgenov) [reply oauth2-server/src/main/java/com/clouway/oauth2/OAuth2ServletApiSupport.java]: It's related to Servlet API as method accepts HttpServletRequest and HttpServletResponse.

A better definition of this interface need to be provided which to not be related to the Servlet API. If it accepts friendly serve requests and responses then the client app which is using the library will be able to create adapter depending on the environment.

Comment (mgenov) [reply oauth2-server/src/main/java/com/clouway/oauth2/OAuth2ServletApiSupport.java]: It could happen in following PR btw.

**Review by mgenov — APPROVED** [review]
LGTM !

**Comment by mgenov** [comment]
1.0.6 is now released and snapshot version of 1.0.7 is now available in sonatype: https://oss.sonatype.org/content/repositories/snapshots/com/clouway/security/oauth2-server/1.0.7-SNAPSHOT/

In an hour it will be available and in maven central.

**Pull request merged**